osu.Game/Overlays/BeatmapSet/BeatmapSetBadgePill.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using JetBrains.Annotations;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public class BeatmapSetBadgePill : CircularContainer
+    {
+        public BeatmapSetBadgePill()
+        {
+            AutoSizeAxes = Axes.Both;
+            Masking = true;
+        }
+
+        [BackgroundDependencyLoader(true)]
+        private void load([CanBeNull] OsuColour colours, [CanBeNull] OverlayColourProvider colourProvider)
+        {
+            Add(new Box
+            {
+                RelativeSizeAxes = Axes.Both,
+                Colour = colourProvider?.Background5 ?? colours?.Gray2 ?? Color4.DarkGray,
+            });
```

Comment:
If this component is to remain, this is a footgun: if implementors aren't careful they can end up nuking the background box on accident by setting `Children`. This should really be a `CompositeDrawable` with `Content` overridden to prevent that.